mcpserver/mcpserver.go

```diff
@@ -0,0 +1,107 @@
+package mcpserver
+
+import (
+	"context"
+
+	"dagger.io/dagger"
+	"github.com/dagger/container-use/environment"
+	"github.com/dagger/container-use/repository"
+	"github.com/mark3labs/mcp-go/mcp"
+)
+
+func dagFromContext(ctx context.Context) *dagger.Client {
+	dag, ok := ctx.Value(daggerClientKey{}).(*dagger.Client)
+	if !ok {
+		panic("dagger client not found in context")
```

Comment:
🐛 **Bug (Critical):** Function panics instead of returning an error, which could crash the entire server. Should return an error that can be handled gracefully.

```suggestion
func dagFromContext(ctx context.Context) (*dagger.Client, error) {
	dag, ok := ctx.Value(daggerClientKey{}).(*dagger.Client)
	if !ok {
		return nil, fmt.Errorf("dagger client not found in context")
	}
	return dag, nil
}
```